src/alfasim_score/converter/alfacase/injection_operation.py

```diff
@@ -20,60 +21,82 @@
 from alfasim_score.constants import WELLBORE_TOP_NODE_NAME
 from alfasim_score.converter.alfacase.base_operation import BaseOperationBuilder
 from alfasim_score.units import FRACTION_UNIT
-from alfasim_score.units import LENGTH_UNIT
-from alfasim_score.units import PRESSURE_UNIT
 from alfasim_score.units import TEMPERATURE_UNIT
-from alfasim_score.units import VELOCITY_UNIT
 
 
 class InjectionOperationBuilder(BaseOperationBuilder):
     def __init__(self, score_filepath: Path):
         super().__init__(score_filepath)
         self.operation_type = OperationType.INJECTION
-        score_operation_type = self.score_input.read_operation_type()
         assert (
-            self.operation_type == score_operation_type
-        ), f"The created operation is injection, but the imported operation is configured as {score_operation_type}."
+            self.general_data["type"] == self.operation_type
+        ), f"The created operation is injection, but the imported operation is configured as {self.general_data['type']}."
+
+    def is_injecting_water(self, alfacase: CaseDescription) -> bool:
+        """Check if the operation has water in the well."""
+        has_inlet_flow = self.general_data["flow_rate"].GetValue() > 0.0
+        has_initial_water = np.any(
+            np.array(
+                alfacase.wells[0].initial_conditions.volume_fractions.table_length.fractions.get(
+                    FLUID_WATER, 0.0
+                )
+            )
+            > 0.0
+        )
+        is_injected_fluid_water = self.general_data["fluid_type"] == FluidType.WATER
+        return (has_initial_water or has_inlet_flow) and is_injected_fluid_water
+
+    def is_injecting_gas(self, alfacase: CaseDescription) -> bool:
+        """Check if the operation has gas in the well."""
+        has_inlet_flow = self.general_data["flow_rate"].GetValue() > 0.0
+        has_initial_water = np.any(
+            np.array(
+                alfacase.wells[0].initial_conditions.volume_fractions.table_length.fractions.get(
```

Comment:
At the end of day, with this check we are always checking `self.general_data["fluid_type"]` twice, isn't?
Because, we set the initial fraction using `self.general_data["fluid_type"]`, thus is this method, we check the initial volume fraction (that used `self.general_data["fluid_type"]`).
Thus, in the next line we check `self.general_data["fluid_type"]` again.
I'm being repetitive, because it seems the code is repetitive 😃

Reply:
Indeed it was duplicated. Changed to just check the type of fluid and it makes sure there is inlet flow.